src/license/license.go

```diff
@@ -59,13 +59,14 @@ func CheckLicenseSecret() error {
 		return errors.New("license secret doesn't exist for discovery-engine")
 	}
 
-	l := &License{
+	LCfg.Lcs = &License{
 		UserId: string(secret.Data["user-id"]),
```

Comment:
What if the secret exists and the required keys `user-id` or `key` is not found.? In this scenario, we can directly return an error and exit right?